dts/arm64/broadcom/bcm2712.dtsi

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 Myeonghyeon Park <myeonghyeon@tsnlab.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm64/armv8-a.dtsi>
+#include <zephyr/dt-bindings/interrupt-controller/arm-gic.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a76";
+			reg = <0>;
+		};
+	};
+
+	interrupt-parent = <&gic>;
+
+	timer {
+		compatible = "arm,armv8-timer";
+		interrupts = <GIC_PPI 13 IRQ_TYPE_LEVEL
+		              IRQ_DEFAULT_PRIORITY>,
+		             <GIC_PPI 14 IRQ_TYPE_LEVEL
+		              IRQ_DEFAULT_PRIORITY>,
+		             <GIC_PPI 11 IRQ_TYPE_LEVEL
+		              IRQ_DEFAULT_PRIORITY>,
+		             <GIC_PPI 10 IRQ_TYPE_LEVEL
+		              IRQ_DEFAULT_PRIORITY>;
+	};
+
+	soc {
+		#address-cells = <2>;
+		#size-cells = <1>;
+
+		sram0: memory@200000 {
+			device_type = "memory";
+			compatible = "mmio-sram";
+			reg = <0x0 0x200000 0x80000>;
+		};
+
+		gic: interrupt-controller@107fff9000 {
+			compatible = "arm,gic-v2", "arm,gic";
+			reg = <0x10 0x7fff9000 0x1000>,
+			      <0x10 0x7fffa000 0x2000>;
+			interrupt-controller;
+			#interrupt-cells = <4>;
+			status = "okay";
+		};
+
+		gpio2@107d517c00 {
+			compatible = "simple-bus";
+			reg = <0x10 0x7d517c00 0x40>;
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+			gio_aon: gpio@0 {
+				compatible = "brcm,brcmstb-gpio";
+				reg = <0>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				ngpios = <17>;
+				status = "disabled";
+			};
+		};
+
+		uart10: serial@107d001000 {
```

Comment:
Why the label is `uart10`?

In Linux case, `uart10` seems as just alias node.

https://github.com/raspberrypi/linux/blob/35f84e1791eaae8c8c0f4798a5cba3061277a672/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts#L175C1-L176C38

https://github.com/raspberrypi/linux/blob/35f84e1791eaae8c8c0f4798a5cba3061277a672/arch/arm/boot/dts/broadcom/bcm2712.dtsi#L186

Reply:
I wrote it like this because I thought that name is what users would expect.

Decompiled dtb showed aliases like this:

```
uart0 = "/axi/pcie@120000/rp1/serial@30000";
...
uart10 = "/soc/serial@7d001000";
```

Also, `dtparam=uart0_console` option in `config.txt` makes the board use UART on GPIO 14 & 15 as console, so I thought that one should be named `uart0`. (and I'm working on it right now)

I considered an option to write exactly what `bcm2712.dtsi` says and make alias nodes in `rpi_5.dts` like what they did in `bcm2712-rpi-5-b.dts` but I thought that would be confusing.

Reply:
OK. I got it.

https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/configuration/uart.adoc#primary-and-secondary-uart